### ISSUE TITLE
Routes not working properly when using Express Routers

### ISSUE THREAD

**Issue by clbranan** [fmcarvalho/express-sitemap-html#8]
Hello! This project is really cool, but for some reason, it does not retain the correct positioning of the '/'s in the url. Example:
Documentation for /api/departments results in /apidepartments///

The Server:
```
const DepartmentRouter = require('./routes/departmentRoutes')
app.use('/api/departments', DepartmentRouter)

const LineRouter = require('./routes/lineRoutes')
app.use('/api/lines', LineRouter)
```

The Department Router:
```
const express = require('express')
const DepartmentController = require('../controllers/departmentController')
const router = express.Router()

router.get('/', DepartmentController.getAllDepartments)
router.get('/html', DepartmentController.getAllDepartmentsHTML)

router.get('/:departmentNumber', DepartmentController.getDepartment)

module.exports = router
```

The Line Router:
```
const express = require('express')
const LineController = require('../controllers/lineController')
const router = express.Router()

router.get('/', LineController.getAllLines)
router.get('/html', LineController.getAllLinesHTML)
router.get('/line', LineController.getLine)
router.get('/department', LineController.getLinesForDepartment)

module.exports = router
```

I've looked a bit through the source, but I'm not seeing exactly where the issue is. Any help would be much appreciated!

![image](https://user-images.githubusercontent.com/77748753/118168183-30116700-b3ed-11eb-94bc-52af50a8c620.png)

**Comment by clbranan** [comment]
I think I've found what's causing the issue, but I now need to find out why. It's with function buildPrev(prev, regexp, path)

```function buildPrev(prev, regexp, path) {
    if(path) return prev + path
    regexp = regexp.toString().replace('/^\\/', '')
    regexp = regexp.replace('?(?=\\/|$)/i', '')
    regexp = regexp.replace('\\/', '')
    regexp = regexp.replace('/^', '')
    return prev + regexp
}```

This is what console.log(app._router.stack) shows:

```...
Layer {
    handle: [Function: router] {
      params: {},
      _params: [],
      caseSensitive: undefined,
      mergeParams: undefined,
      strict: undefined,
      stack: [Array]
    },
    name: 'router',
    params: undefined,
    path: undefined,
    keys: [],
    regexp: /^\/api\/departments\/?(?=\/|$)/i {
      fast_star: false,
      fast_slash: false
    },
    route: undefined
  }
...```

**Comment by clbranan** [comment]
I am fairly new to git, so I am not sure how to make a pull request exactly. Regardless, I have fixed it for my scenario:

```
function buildPrev(prev, regexp, path) {
    if(path) return prev + path
    regexp = regexp.toString().replace('/^\\/', '')
    regexp = regexp.replace('?(?=\\/|$)/i', '')
    regexp = regexp.replace(/(\\\/)/g, '/') // replaces all occurrences of "\\/" with "/"
    regexp = regexp.replace(/(\/)$/g, '') // replaces last occurrence of "/" with "" IF the URL ends with a "/", as something else puts a "/" at the end of the url.
    regexp = regexp.replace('/^', '')
    return prev + regexp
}
```

**Comment by fmcarvalho** [comment]
I have tested your use case in the demo application of [example](https://github.com/fmcarvalho/express-sitemap-html/tree/master/example) folder and I have observed that erroneous behavior. 

And your patch it seems to solve the problem and itis still compliant with current unit tests.

Feel free to make a PR with that fix, but please also update the unit test `Test parse routes` to check your use case.

To make a PR you first need to fork this repo to your Github account. After that you can make those fixes in our repo and at the end, you use the Github user interface to make the PR from your forked repo to this one. 

Best regards,
Miguel

**Comment by fmcarvalho** [comment]
Fixed on release 1.2.5
